python/cuml/linear_model/logistic_regression.pyx

```diff
@@ -314,6 +337,38 @@ class LogisticRegression(Base):
         """
         return self.qn.predict(X, convert_dtype=convert_dtype)
 
+    def predict_proba(self, X, convert_dtype=False):
+        """
+        Predicts the class probabilities for X
+
+        Parameters
+        ----------
+        X : array-like (device or host) shape = (n_samples, n_features)
+            Dense matrix (floats or doubles) of shape (n_samples, n_features).
+            Acceptable formats: cuDF DataFrame, NumPy ndarray, Numba device
+            ndarray, cuda array interface compliant array like CuPy
+
+        convert_dtype : bool, optional (default = False)
+            When set to True, the predict method will, when necessary, convert
+            the input to the data type which was used to train the model. This
+            will increase memory used for the method.
+        Returns
+        ----------
+        y: cuDF DataFrame
+           Dense matrix (floats or doubles) of shape (n_samples, C)
+        """
+        proba = self.decision_function(X, convert_dtype=convert_dtype)
+        proba = cp.asarray(proba.to_gpu_matrix())
+        if self.num_classes == 2:
+            proba = 1 / (1 + cp.exp(-proba))
+            proba = cp.column_stack((proba, proba))
```

Comment:
I would preallocate proba to be the right size then fill in its columns. Feels weird that here we briefly have a scrambled/incorrect proba before fixing in the next line.